### ISSUE TITLE
Pagetitle simulator returns TYPO3's 404 page for hidden pages

### ISSUE THREAD

**Issue by thomaszbz** [mblaschke/TYPO3-metaseo#93]
As of current develop (dc5c8d4c053f42e64d735180bbcdafdfbce7b6ce), when calling the list for

```
Pagetitle Simulator (SLOW!)
```

the ajax response contains the TYPO3 CMS 404 page with http status 404.

The url for the ajax method call is

```
https://example.com/typo3/ajax.php?ajaxID=tx_metaseo_backend_ajax::page&cmd=getList
```

All the other lists like 'Keywords' and 'Pagetitle' run fine on the same TYPO3 CMS instance.

Even if processing hidden pages would not be supported, then the output of TYPO3's 404 page would not be suitable for an ajax response. Therefore, this issue can be classified as a bug in any case.

Note that the  #92 is not merged yet into develop and therefore cannot be the reason for this bug.

**Comment by thomaszbz** [comment]
It's not the routing (pagetitle uses the same url!). However, for pagetitlesim, the 404 happens in PageAjax->listPageTitleSim() because debug code

    $list = $this->listPageTitleSim($page, $depth, $sysLanguage);
    return $this->ajaxError('', '', 401);

returns http status code 404 while

    return $this->ajaxError('', '', 401);
    $list = $this->listPageTitleSim($page, $depth, $sysLanguage);

returns http status code 401.

**Comment by thomaszbz** [comment]
Further tracked it down as before:

The 404-error is thrown by PageAjax->simulateTitle in line

    $this->initTsfe($page, null, $page, null, $sysLanguage);

with parameters:

- $page = a page array with data of a valid page which is however hidden
- $sysLanguage = 0

**Comment by thomaszbz** [comment]
Further tracked it down, now it's

    FrontendUtility::init($page['uid'], $rootLine, $pageData, $rootlineFull, $sysLanguage);

in PageAjax->initTsfe()

**Comment by thomaszbz** [comment]
Further tracked it down, now it's

    $GLOBALS['TSFE']->determineId();

in \Metaseo\Metaseo\Utility\FrontendUtility->init() which is throwing the 404 page on hidden pages.

**Comment by thomaszbz** [comment]
Related: #210 . Fix for #93 might look similar.

**Comment by thomaszbz** [comment]
Retried with TYPO3 7.6.14, Metaseo 2.0.2:

The pagetitle simulator now shows all entries. However, for hidden subpages, pagetitle simulator shows the title of the parent page:

![arbeitsflache 1_105](https://cloud.githubusercontent.com/assets/11200727/21575022/b1c54398-ceff-11e6-8525-275a398252c1.png)

**Comment by thomaszbz** [comment]
In TYPO3 CMS 6.2.29, I still get the 404 error, together with Metaseo 2.0.2

**Comment by thomaszbz** [comment]
fdde9dc already is valid for 7.6.14, but not for 6.2.29.

6.2.29 still throws the 404 error.

**Comment by thomaszbz** [comment]
Now that it works with 7.6.14 using functionality which is available in 6.2.29 as well, it looks as if 6.2.29 has a buggy implementation which somehow got fixed up until 7.6.14.

It really seems hard to somehow fix this for 6.2.29, in case it's a bug in the core. Using XClasses again also is ugly.

Now that nobody ever complained about this issue, I won't fix the bug for 6.2.29, which is EOL soon anyways.

The bug is fixed for 7.6.14, however.
